### DISCUSSION TITLE
Feature request: build pure python wheel for PyPi

### DISCUSSION THREAD

**Discussion by johncalvinyoung** [mozman/ezdxf#774] [Ideas]
I'm working with your package (thanks so much!) doing some DXF export from a drawing application running in the browser. I've been able to get the core ezdxf package working with Pyodide (Python running in WASM), but it required building a vanilla python wheel from source, as the PyPi builds don't have a py3-none-any.whl build since 0.14.x. Could you look into building that for the package repository? It'd be great to get a further ported version with the extensions built for the [Pyodide (WASM/Emscripten) environment](https://pyodide.org/en/stable/development/new-packages.html), but even just the pure core package would be really helpful.

Thanks so much!

**Comment by mozman** [comment]
Why is the source package not good enough?

![image](https://user-images.githubusercontent.com/4230966/200741542-dc9f4255-0f71-4b73-a903-472349485363.png)

Exist a dedicated `setup.py` command like `bdist_wheel` to build such a package? The only way I found to build such a packages is using `pypy` which doesn't use the c-extensions.

If you want special packages, look at the [workflow](https://github.com/mozman/ezdxf/tree/master/.github/workflows) scripts and create a similar one that is simple enough that I understand it, because I have to maintain it. All wheels are build automatically by a github-action using the [cibuildwheel](https://github.com/pypa/cibuildwheel) server provided by the Python Packaging Authority.

**Reply by johncalvinyoung** [reply]
Thanks for the reply, and sorry for not getting back to you quicker--been very busy implementing features on top of ezdxf for my application!

Sadly, `micropip`, which is the package manager for the Pyodide ecosystem, is really really picky about loading wheels. I attempted to pass it the link to the 1.0.0.zip file on PyPi and it barfed.

Guess I'll look into contributing a build script.

**Comment by mozman** [comment]
I added a pure Python wheel deployer by using pypy to run the setup.py script, which does not build the C-extensions:

![image](https://user-images.githubusercontent.com/4230966/208226037-ba0bbb05-994a-4d50-a491-d81cb0c5b2b6.png)

use `python3 -m pip install ezdxf==1.0.1b0`

**Reply by johncalvinyoung** [reply]
Thank you so much!
